qemu/tests/boot_order_check.py

```diff
@@ -74,25 +75,20 @@ def run(test, params, env):
                                          list_nic_addr[1][0],
                                          list_nic_addr[2][0])
 
-    error.context("Check the guest boot result", logging.info)
-    if bootorder_type == "type2":
-        session_serial = vm.wait_for_serial_login(timeout=timeout)
-        output = vm.serial_console.get_stripped_output()
-        session_serial.close()
-    else:
-        output = vm.serial_console.get_stripped_output()
-        utils_misc.wait_for(lambda: re.search("No bootable device.", output),
-                            timeout, 1)
+    error_context.context("Check the guest boot result", logging.info)
+    start = time.time()
+    while True:
         output = vm.serial_console.get_stripped_output()
-
-    # find and replace some ascii characters to non-ascii char,
-    # like as: '\b' (backspace)
-    if backspace_char:
-        data = re.sub(r".%s" % backspace_char, "", output)
-    else:
-        data = output
-    result = re.findall(boot_fail_infos, data, re.S | re.M | re.I)
-
+        # find and replace some ascii characters to non-ascii char,
+        # like as: '\b' (backspace)
+        if backspace_char:
+            data = re.sub(r".%s" % backspace_char, "", output)
+        else:
+            data = output
+        result = re.findall(boot_fail_infos, data, re.S | re.M | re.I)
+        if result or time.time() > start + timeout:
+            break
+        time.sleep(1)
     if not result:
-        raise error.TestFail("Got a wrong boot order, "
-                             "Excepted order: '%s'" % bootorder)
+        test.error("Timeout when try to get expected boot order: "
+                   "'%s' " % bootorder)
```

Comment:
Better to report the actual result.

Reply:
Thanks, print the `output` log for failure debug.